uwuw/uwuw_preprocessor.cpp

```diff
@@ -0,0 +1,693 @@
+{
+  // make new name concatenator class
+  ncr = new name_concatenator();
+
+  // load the materials
+  material_library = mat_lib.load_pyne_materials(material_library_filename,"/material_library/materials");
+
+  // load the material objects
+  // load the dag file
+  moab::ErrorCode rval = DAG->load_file(dagmc_filename.c_str());
+
+  // do the minimal DAGMC initialisation
+  rval = DAG->setup_impl_compl();
+  rval = DAG->setup_indices();
+
+  // set the output filename
+  output_filename = output_file;
+
+  // set the verbosity
+  verbose = verbosity;
+  fatal = fatal_errors;
+}
+
+// destructor
+uwuw_preprocessor::~uwuw_preprocessor()
+{
+}
+
+// get the dagmc properties from the geometry file
+std::map<moab::EntityHandle, std::vector<std::string> > get_property_assignments( std::string property,
+    int dimension,
+    std::string delimiters)
+{
+  std::map<moab::EntityHandle,std::vector<std::string> > prop_map; // to return
+  std::vector<std::string> test_keywords; // keywords we are going to test against
+  test_keywords.push_back(property);
+  std::map<std::string, std::string> keyword_synonyms; // empty synonynms map
+
+  // get initial sizes
+  int num_entities = DAG->num_entities( dimension );
+
+  // parse data from geometry
+  moab::ErrorCode rval = DAG->parse_properties( test_keywords, keyword_synonyms,delimiters.c_str());
+
+  if (moab::MB_SUCCESS != rval) {
+    std::cerr << "DAGMC failed to parse metadata properties" <<  std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  // loop over all cells
+  for( int i = 1; i <= num_entities; ++i ) {
+    std::vector<std::string> properties; //vector of properties coming back
+
+    // get cellid
+    moab::EntityHandle entity = DAG->entity_by_index( dimension, i );
+
+    // get the group contents
+    if( DAG->has_prop( entity, property ) )
+      rval = DAG->prop_values(entity,property,properties);
+    else
+      properties.push_back("");
+
+    // remove duplicates
+    std::vector<std::string>::iterator it;
+    it = std::unique(properties.begin(),properties.end());
+    // resize vector to remove empty parts
+    properties.resize(std::distance(properties.begin(),it));
+
+    // assign the map value
+    prop_map[entity]=properties;
+
+  }
+
+  return prop_map;
+}
+
+// write the new material library
+void uwuw_preprocessor::write_uwuw_materials()
+{
+  std::map<std::string,pyne::Material> :: iterator it;
+
+  // loop over the processed material library and write each one to the file
+  for ( it = uwuw_material_library.begin() ; it != uwuw_material_library.end() ; ++it ) {
+    // the current material
+    pyne::Material mat = it->second;
+    // write the material to the file
+    if(verbose) {
+      std::cout << "writing material, " << mat.metadata["name"].asString();
+      std::cout << "writing material, " << mat.metadata["fluka_name"].asString();
+      std::cout << " to file " << output_filename << std::endl;
+    }
+
+    mat.write_hdf5(output_filename,"/materials");
```

Comment:
For the input file, the hdf5 path is "/material_library/materials", which is inconsistant with this line. I am guessing there is a reason for this. If there isn't, the PyNE default of "/materials" should probably be always used.